### PR TITLE
Auto release when image changes

### PR DESCRIPTION
**Pull request by lgc2333**
# Pull Request

## Description

[Please include a brief description of the changes or features you are proposing.]

## Related Issues

- #378 
- #616 

## Pictures (if applicable)

[Include any pictures that help demonstrate the changes.]

## Checklist

- [x] I have tested the changes/updates before submitting this pull request.  

## Additional Notes

Preview:

https://github.com/lgc2333/ServiceLogos/releases  
https://github.com/lgc2333/ServiceLogos/tree/test-auto-release

### CODE REVIEW

**Comment by lgc2333** [comment]
i am not familiar with bash, i'll write a python script to generate release message, wait me sometime

**Comment by lgc2333** [comment]
okay for merge
i'm tired, i'll go sleep now
good night~~~~~~

File: docs/scripts/generate-changelog.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+def decode(s: bytes) -> str:
+    try:
+        return s.decode()
+    except UnicodeDecodeError:
+        return s.decode(ENC, errors="replace")
+
+
+def run(cmd: list[str]) -> str:
+    out = decode(Popen(cmd, stdout=PIPE).communicate()[0])  # noqa: S603
+    # print(out)
```
Comment (VVsxmja) [inline docs/scripts/generate-changelog.py:25]: delete this line 🤔

Comment (VVsxmja) [inline docs/scripts/generate-changelog.py:25]: This line should be deleted. 🤔

File: docs/scripts/generate-changelog.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        return
+
+    changelog = "\n".join(
+        [
+            f"- **{NAME.get(status[0])}**: {' -> '.join(f'`{x}`' for x in rest)}"
+            for status, *rest in diff_spiltted
+        ],
+    )
+    print(changelog)
+    Path("changelog.md").write_text(changelog, "u8")
```
Comment (VVsxmja) [inline docs/scripts/generate-changelog.py:68]: `u8` is an alias. `utf_8` should be better. [Reference](https://docs.python.org/3/library/codecs.html#standard-encodings)

File: .github/workflows/auto-release.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 2
+          submodules: true
+      - name: Run Commands
```
Comment (VVsxmja) [inline .github/workflows/auto-release.yml:20]: I think it should be better to provide a descriptive name here, such as `Generate Changelog`.

**Review by VVsxmja — CHANGES_REQUESTED** [review]

File: docs/scripts/generate-changelog.py (hunk 2 above)
Comment (lgc2333) [reply docs/scripts/generate-changelog.py:68]: different person has different code styles, i'll keep this

**Review by T-taku — APPROVED** [review]
lgtm;)

**Pull request merged**